### ISSUE TITLE
Library artifacts are published to jcenter only, which will close in 3 months

### ISSUE THREAD

**Issue by entibaldi** [gojuno/koptional#40]
Hello! Your library is published only to jcenter. Please consider publishing it also to mavenCentral, as jcenter is [shutting down in May](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/).

[Here](https://central.sonatype.org/pages/producers.html) is some info on how to publish artifacts to mavenCentral.

**Comment by arturdryomov** [comment]
Thanks for the automated message 😉 We are aware of the issue, will look into Maven Central publishing.

**Comment by welsinga** [comment]
Any update on this?

**Comment by arturdryomov** [comment]
Yep, thanks for the reminder. The main pain-point here is that Juno does not exist for a long time (RIP). Meaning that the `gojuno.com` domain is dead as well. I’ve been in multiple private and public discussions about the Maven Central publication without owning the domain but it’s impossible. Meaning that we’ll have to use a `com.github.gojuno` group name instead, breaking the compatibility. So yeah, the re-publication needs to be done and technically this can proceed but I’ve spent some time on discussions and pinging people so far.

**Comment by artem-zinnatullin** [comment]
Alternatively it can be moved to something w/o Juno in the name but with
credits to the origin company Juno in readme, I don’t think anyone will be
against..

You could own that Maven group name since you’re most active maintainer

**Comment by arturdryomov** [comment]
@artem-zinnatullin, yeah, like Lyft 🦀

**Comment by artem-zinnatullin** [comment]
I mean that works :)

**Comment by luhmirin-s** [comment]
Any update?

**Comment by arturdryomov** [comment]
WIP! FYI — [JFrog reversed the ship once again](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/).

**Comment by duprass** [comment]
Any update?

**Comment by tonnyavery** [comment]
We are switching our project off jcenter already, any update on koptional availability somewhere else?

**Comment by tonnyavery** [comment]
FYI, this link now gives 403 https://jcenter.bintray.com/com/gojuno/koptional

**Comment by radhey-telcoin** [comment]
Any update on this?

**Comment by andro-jedi** [comment]
Just copy these few files to your project =)

**Comment by MartinSadovy** [comment]
now it is gone :D

**Comment by arturdryomov** [comment]
It isn’t but [the JCenter is down](https://stackoverflow.com/questions/74258160/is-jcenter-down-permanently-31-oct) for unknown reason and for unknown time. FYI — [the read-scoped repo should’ve been available indefinitely](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/). It’s unknown if this period became definite.

**Comment by arturdryomov** [comment]
Re-iterating over the publishing issue. As I’ve mentioned, Juno doesn’t exist. Same with the JCenter publishing. Artifacts need to be published at Maven Central. There are two approaches for this.

* As `com.github.gojuno.koptional`. Imports remain the same, artifact IDs will be different. Reusing the `com.gojuno.koptional` group ID is impossible. I’ve bugged Sonatype from multiple angles and it’s impossible to gain access to a domain group ID without an option to prove the domain ownership — and there is no such option since Juno is gone. The domain expires on 2024.09. However, according to [the WHOIS](https://who.is/whois/gojuno.com), Lyft owns it. @artem-zinnatullin, do you want to dig into this?
* As `com.corp.koptional`. Imports most likely will be different, artifact IDs will be different. Original developers work at Lyft now so this sounds like a good choice. However, the import change gonna break a lot of code. Unless we publish 1.7.0 as is and 2.0.0 with new imports. I’m gonna leave this to @artem-zinnatullin as well.

**Comment by nokite** [comment]
@arturdryomov thank you for all the efforts you've put into this. 
Any of the solutions discussed seem OK to me. In any case they're a better alternative to not having this on Maven Central.

Even changing the imports manually is fine. As long as it's a simple text replace that's consistent among all the imports, I can do it in seconds over the whole project, probably by using regular expressions and "replace all".
